cocos/base/CCUserDefault.cpp

```diff
@@ -407,76 +243,71 @@ void UserDefault::setDelegate(UserDefault *delegate)
     _userDefault = delegate;
 }
 
-
-bool UserDefault::isXMLFileExist()
-{
-    return FileUtils::getInstance()->isFileExist(_filePath);
-}
-
-void UserDefault::initXMLFilePath()
+void UserDefault::init()
 {
-    if (! _isFilePathInitialized)
+    if (! _initialized)
     {
-        _filePath += FileUtils::getInstance()->getWritablePath() + XML_FILE_NAME;
-        _isFilePathInitialized = true;
-    }    
-}
-
-// create new xml file
-bool UserDefault::createXMLFile()
-{
-    bool bRet = false;  
+        _filePath = FileUtils::getInstance()->getWritablePath() + USER_DEFAULT_FILENAME;
 
-    pugi::xml_document doc;
-    doc.load_string(R"(<?xml version="1.0" encoding="UTF-8" ?>)"
-"<" USERDEFAULT_ROOT_NAME " />", pugi::parse_full);
+        // construct file mapping
+        _fd = posix_open(_filePath.c_str(), O_WRITE_FLAGS);
+        if (_fd == -1) {
+            log("[Warnning] UserDefault::init open storage file '%s' failed!", _filePath.c_str());
+            return;
+        }
 
-    bRet = doc.save_file(_filePath.c_str(), " ");
+        int filesize = posix_lseek(_fd, 0, SEEK_END);
+        posix_lseek(_fd, 0, SEEK_SET);
 
-    return bRet;
-}
+        if (filesize < _curMapSize) {
+            posix_fsetsize(_fd, _curMapSize);
+        }
+        /// load to memory _values
+        _rwmmap = std::make_shared<mio::mmap_sink>(posix_fd2fh(_fd), 0, _curMapSize);
+        if (_rwmmap->is_mapped()) { // no error
+            yasio::ibstream_view ibs(_rwmmap->data(), _rwmmap->length());
+
+            if (ibs.length() > 0) {
+                // count
+                // key/value --> string/string
+                int count = ibs.read_i7();
+                for (auto i = 0; i < count; ++i) {
+                    auto key = ibs.read_v();
+                    auto value = ibs.read_v();
+                    this->_values.emplace(key, value);
+                }
+            }
+        }
 
-const string& UserDefault::getXMLFilePath()
-{
-    return _filePath;
+        _initialized = true;
+    }    
 }
 
 void UserDefault::flush()
 {
-}
-
-void UserDefault::deleteValueForKey(const char* key)
-{
-    pugi::xml_node rootNode;
-    pugi::xml_document doc;
-    pugi::xml_node node;
-
-    // check the params
-    if (!key)
-    {
-        CCLOG("the key is invalid");
-        return;
+    // TODO: Win32 store to UserDefault.plain file?
```

Comment:
For win32, indeed UserDefault.plain could be better, especially better for use to debug. may be put it in next enhancement.

Reply:
Yeah, I will continue this PR later.